### PR TITLE
Release 6.18.2

### PR DESCRIPTION
**Pull request by LukasTy**
Don't mind the incorrect branch name. 🙈 😭 

### Release the packages

- [ ] Checkout the last version of the working branch
- [ ] Make sure you have the latest dependencies installed: `yarn`.
- [ ] Build the packages: `yarn release:build`.
- [ ] Release the versions on npm: `yarn release:publish` (you need your 2FA device).
- [ ] Push the newly created tag: `yarn release:tag`.

### Publish the documentation

The documentation must be updated on the `docs-vX` branch (`docs-v4` for `v4.X` releases, `docs-v5` for `v5.X` releases, ...)

- [ ] Push the working branch on the documentation release branch to deploy the documentation with the latest changes.

```bash
git push upstream master:docs-v6 -f
```

You can follow the deployment process [on the Netlify Dashboard](https://app.netlify.com/sites/material-ui-x/deploys?filter=docs-v6)
Once deployed, it will be accessible at https://material-ui-x.netlify.app/ for the `docs-v6` deployment.

### Publish GitHub release

- [ ] After documentation is deployed, publish a new release on [GitHub releases page](https://github.com/mui/mui-x/releases)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,59 @@
...
+Same changes as in `@mui/x-date-pickers@6.18.2`.
+
+### Charts / `@mui/x-charts@6.18.2`
+
+- [charts] Fix `ChartsTooltip` component setup (#11157) @LukasTy
+- [charts] Remove outdated prop-types (#10998) @alexfauquette
+
+### Docs
+
+- [docs] Fix Pickers CustomComponents Incorrect naming of a component in example (#11003) @lhilgert9
```
Comment (flaviendelangle) [inline CHANGELOG.md:55]:
```suggestion
- [docs] Fix incoherent naming of a component in the *Custom component* page (#11003) @lhilgert9
```

**Review by flaviendelangle — APPROVED** [review]
Looks solid, nice localization improvements this week

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,59 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 6.18.2
+
+_Nov 23, 2023_
+
+We'd like to offer a big thanks to the 12 contributors who made this release possible. Here are some highlights ✨:
```
Comment (LukasTy) [inline CHANGELOG.md:10]:
```suggestion
We'd like to offer a big thanks to the 11 contributors who made this release possible. Here are some highlights ✨:
```

File: CHANGELOG.md (hunk 1 above)
Comment (LukasTy) [reply CHANGELOG.md:55]:
```suggestion
- [docs] Fix incoherent naming of a component in `Custom slots and subcomponents` page (#11003) @lhilgert9
```
I am going with regular marking as we seem to use italics or bold rarely.
Some of the latest examples of mentioned pages are also wrapped in ``. 🙈 
Also updated the page name to be coherent with the existing page.

**Pull request merged**